[Bug] [seatunnel-connector-flink-jdbc] mysql query matcher tablename error

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

mysql to hdfs ，but jdbc source query not support `where ....` condition

### SeaTunnel Version

a7ff0f8805a063a7c7a33f256f82f28235f8e1b1

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
}

source {

  JdbcSource {
     driver = "com.mysql.jdbc.Driver"
     url = "jdbc:mysql://10.202.254.10:3306/test?useUnicode=true&characterEncoding=utf8&useSSL=false"
     username = "root"
     password = "123456"
     query = "select username,admin from dc_user where admin=1"
     result_table_name = "user"
  }

}

transform {

}
sink {
  ConsoleSink {}
}
```


### Running Command

```shell
idea run LocalFlinkExample
```


### Error Exception

```log
22/03/18 10:23:08 ERROR Seatunnel: Fatal Error, 

22/03/18 10:23:08 ERROR Seatunnel: Please submit bug report in https://github.com/apache/incubator-seatunnel/issues

22/03/18 10:23:08 ERROR Seatunnel: Reason:null 

22/03/18 10:23:08 ERROR Seatunnel: Exception StackTrace:java.lang.NullPointerException
	at org.apache.flink.api.java.typeutils.TupleTypeInfoBase.<init>(TupleTypeInfoBase.java:64)
	at org.apache.flink.api.java.typeutils.RowTypeInfo.<init>(RowTypeInfo.java:89)
	at org.apache.seatunnel.flink.source.JdbcSource.getRowTypeInfo(JdbcSource.java:169)
	at org.apache.seatunnel.flink.source.JdbcSource.prepare(JdbcSource.java:129)
	at org.apache.seatunnel.flink.source.JdbcSource.prepare(JdbcSource.java:59)
	at org.apache.seatunnel.command.BaseTaskExecuteCommand.lambda$prepare$0(BaseTaskExecuteCommand.java:72)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.apache.seatunnel.command.BaseTaskExecuteCommand.prepare(BaseTaskExecuteCommand.java:72)
	at org.apache.seatunnel.command.flink.FlinkTaskExecuteCommand.execute(FlinkTaskExecuteCommand.java:49)
	at org.apache.seatunnel.command.flink.FlinkTaskExecuteCommand.execute(FlinkTaskExecuteCommand.java:35)
	at org.apache.seatunnel.Seatunnel.run(Seatunnel.java:48)
	at org.apache.seatunnel.example.flink.LocalFlinkExample.main(LocalFlinkExample.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.CommandLineWrapper.main(CommandLineWrapper.java:66)
 
22/03/18 10:23:08 ERROR Seatunnel: 
===============================================================================



Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.CommandLineWrapper.main(CommandLineWrapper.java:66)
Caused by: java.lang.NullPointerException
	at org.apache.flink.api.java.typeutils.TupleTypeInfoBase.<init>(TupleTypeInfoBase.java:64)
	at org.apache.flink.api.java.typeutils.RowTypeInfo.<init>(RowTypeInfo.java:89)
	at org.apache.seatunnel.flink.source.JdbcSource.getRowTypeInfo(JdbcSource.java:169)
	at org.apache.seatunnel.flink.source.JdbcSource.prepare(JdbcSource.java:129)
	at org.apache.seatunnel.flink.source.JdbcSource.prepare(JdbcSource.java:59)
	at org.apache.seatunnel.command.BaseTaskExecuteCommand.lambda$prepare$0(BaseTaskExecuteCommand.java:72)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.apache.seatunnel.command.BaseTaskExecuteCommand.prepare(BaseTaskExecuteCommand.java:72)
	at org.apache.seatunnel.command.flink.FlinkTaskExecuteCommand.execute(FlinkTaskExecuteCommand.java:49)
	at org.apache.seatunnel.command.flink.FlinkTaskExecuteCommand.execute(FlinkTaskExecuteCommand.java:35)
	at org.apache.seatunnel.Seatunnel.run(Seatunnel.java:48)
	at org.apache.seatunnel.example.flink.LocalFlinkExample.main(LocalFlinkExample.java:34)
	... 5 more
```


### Flink or Spark Version

flink 1.13.6

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java
+++ b/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java
@@ -25,8 +25,6 @@ import static org.apache.seatunnel.flink.Config.SOURCE_FETCH_SIZE;
 import static org.apache.seatunnel.flink.Config.URL;
 import static org.apache.seatunnel.flink.Config.USERNAME;
 
-import org.apache.flink.api.java.tuple.Tuple;
-import org.apache.flink.api.java.tuple.Tuple2;
 import org.apache.seatunnel.common.config.CheckConfigUtil;
 import org.apache.seatunnel.common.config.CheckResult;
 import org.apache.seatunnel.flink.FlinkEnvironment;
@@ -42,6 +40,7 @@ import org.apache.seatunnel.shade.com.typesafe.config.Config;
 import org.apache.flink.api.common.typeinfo.TypeInformation;
 import org.apache.flink.api.java.DataSet;
 import org.apache.flink.api.java.operators.DataSource;
+import org.apache.flink.api.java.tuple.Tuple2;
 import org.apache.flink.api.java.typeutils.RowTypeInfo;
 import org.apache.flink.types.Row;
 import org.slf4j.Logger;

--- a/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java
+++ b/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java
@@ -71,7 +71,7 @@ public class JdbcSource implements FlinkBatchSource {
     private int fetchSize = DEFAULT_FETCH_SIZE;
     private Set<String> fields;
 
-    private static final Pattern COMPILE = Pattern.compile("select (.+) from (.+).*");
+    private static final Pattern COMPILE = Pattern.compile("[\\s]*select[\\s]*(.*)from[\\s]*([\\S]+).*");
 
     private JdbcInputFormat jdbcInputFormat;
 
@@ -126,7 +126,10 @@ public class JdbcSource implements FlinkBatchSource {
             fetchSize = config.getInt(SOURCE_FETCH_SIZE);
         }
 
-        jdbcInputFormat = JdbcInputFormat.buildFlinkJdbcInputFormat().setDrivername(driverName).setDBUrl(dbUrl).setUsername(username).setPassword(password).setQuery(query).setFetchSize(fetchSize).setRowTypeInfo(getRowTypeInfo()).finish();
+        jdbcInputFormat = JdbcInputFormat.buildFlinkJdbcInputFormat()
+                .setDrivername(driverName).setDBUrl(dbUrl).setUsername(username)
+                .setPassword(password).setQuery(query).setFetchSize(fetchSize)
+                .setRowTypeInfo(getRowTypeInfo()).finish();
     }
 
     private RowTypeInfo getRowTypeInfo() {

--- a/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java
+++ b/seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java
@@ -25,6 +25,8 @@ import static org.apache.seatunnel.flink.Config.SOURCE_FETCH_SIZE;
 import static org.apache.seatunnel.flink.Config.URL;
 import static org.apache.seatunnel.flink.Config.USERNAME;
 
+import org.apache.flink.api.java.tuple.Tuple;
+import org.apache.flink.api.java.tuple.Tuple2;
 import org.apache.seatunnel.common.config.CheckConfigUtil;
 import org.apache.seatunnel.common.config.CheckResult;
 import org.apache.seatunnel.flink.FlinkEnvironment;
@@ -106,19 +108,9 @@ public class JdbcSource implements FlinkBatchSource {
         dbUrl = config.getString(URL);
         username = config.getString(USERNAME);
         String query = config.getString(QUERY);
-        Matcher matcher = COMPILE.matcher(query);
-        if (matcher.find()) {
-            String var = matcher.group(1);
-            tableName = matcher.group(2);
-            if (!"*".equals(var.trim())) {
-                LinkedHashSet<String> vars = new LinkedHashSet<>();
-                String[] split = var.split(",");
-                for (String s : split) {
-                    vars.add(s.trim());
-                }
-                fields = vars;
-            }
-        }
+        Tuple2<String, Set<String>> tableNameAndFields = getTableNameAndFields(COMPILE, query);
+        tableName = tableNameAndFields.f0;
+        fields = tableNameAndFields.f1;
         if (config.hasPath(PASSWORD)) {
             password = config.getString(PASSWORD);
         }
@@ -130,6 +122,27 @@ public class JdbcSource implements FlinkBatchSource {
                 .setDrivername(driverName).setDBUrl(dbUrl).setUsername(username)
                 .setPassword(password).setQuery(query).setFetchSize(fetchSize)
                 .setRowTypeInfo(getRowTypeInfo()).finish();
+    }
+
+    private Tuple2<String, Set<String>> getTableNameAndFields(Pattern regex, String selectSql) {
+        Matcher matcher = regex.matcher(selectSql);
+        String tableName;
+        Set<String> fields = null;
+        if (matcher.find()) {
+            String var = matcher.group(1);
+            tableName = matcher.group(2);
+            if (!"*".equals(var.trim())) {
+                LinkedHashSet<String> vars = new LinkedHashSet<>();
+                String[] split = var.split(",");
+                for (String s : split) {
+                    vars.add(s.trim());
+                }
+                fields = vars;
+            }
+            return new Tuple2<>(tableName, fields);
+        } else {
+            throw new IllegalArgumentException("can't find tableName and fields in sql :" + selectSql);
+        }
     }
 
     private RowTypeInfo getRowTypeInfo() {

